### PR TITLE
Modified tests extensions from .JS to .TSX

### LINKED ISSUE #1543: Change components tests extension from js to jsx or tsx
**Is your feature request related to a problem? Please describe.**
Now the tests are `.js` extension. Inside of them, we have `jsx` code. We should change their extension to `jsx` or `tsx`.

**Describe the solution you'd like**
Change the tests extension to `jsx` or `tsx`.

### PR DESCRIPTION
**Pull request by Mil4n0r**
**Checklist**
_(Check off all the items before submitting)_

- [ ] Build process is done without errors. All tests pass in the `/lib` directory.
- [ ] Self-reviewed the code before submitting.
- [ ] Meets accessibility standards.
- [ ] Added/updated documentation to `/website` as needed.
- [ ] Added/updated tests as needed.

**Description**
In the last introduced Components, we were already using tsx extension for tests. It was a good idea to convert all the existing functional and accessibility tests to tsx in order to improve consistency.

**Closes #1543**

### CODE REVIEW

**Review by GomezIvann — CHANGES_REQUESTED** [review]

File: lib/src/table/types.ts [hunk 1]
```diff
@@ -20,10 +20,10 @@ export type ActionCellsPropsType = {
         onClick: () => void;
         disabled?: boolean;
         tabIndex?: number;
-        options?: never;
+        options?: never | undefined;
       }
     | {
-        icon?: never;
+        icon?: never | undefined;
```
[the remark this answers is not in the document]
Comment (Mil4n0r) [reply lib/src/table/types.ts:26]: Yes, I thought the same but for some reason, when not adding undefined as an option, there was a TS problem saying that "never" was not matching "undefined" type, I will take another look at it.

[the remark this answers is not in the document]
Comment (Mil4n0r) [reply lib/src/table/types.ts]: It seems that there was a problem with the type inference from TS, I did a workaround and now it seems to be solved.

[the remark this answers is not in the document]
Comment (Mil4n0r) [reply lib/src/table/types.ts]: It seems to be a problem regarding TS type inference. I am not sure about how to solve it without manually typing the `actions` for both tests, this should not be needed...

image

[the remark this answers is not in the document]
Comment (Mil4n0r) [reply lib/src/table/types.ts]: As the problem only happens in the test files, I have added the type manually to prevent wrong TS type inference.

**Review by GomezIvann — DISMISSED** [review]

**Review by GomezIvann — APPROVED** [review]

**Pull request merged**